### PR TITLE
added referred_to_external field PatientConsultationDischargeSerializer

### LINKED ISSUE #1429: PatientConsultationDischargeSerializer is not having referred_to_external field.
**Describe the bug**
`PatientConsultationDischargeSerializer` is not having `referred_to_external` field due to which it is not updated in the database.

**To Reproduce**
1. Go to [PR](https://github.com/coronasafe/care_fe/pull/5730) and run the code in that branch.
2. Discharge patient with the reason referred.
3. The postgres database will not update the` referred_to_external` field since `PatientConsultationDischargeSerializer `does not have it.

**Expected behavior**
The database should have the name to the facility referred to.

### PR DESCRIPTION
**Pull request by sachdevavaibhav**
added referred_to_external field PatientConsultationDischargeSerializer

### Associated Issue

- fixes #1429 
- required by coronasafe/care_fe#5407

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins @rithviknishad

### CODE REVIEW

**Comment by sachdevavaibhav** [comment]
@rithviknishad Kindly review it.

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@sachdevavaibhav lint CI is failing, make sure pre-commit is installed and ensure lints passes. 

https://care-be-docs.coronasafe.network/local-setup/configuration.html#setting-up-pre-commit

**Comment by sachdevavaibhav** [comment]
@rithviknishad Done with formatting.

**Review by rithviknishad — APPROVED** [review]

File: care/facility/api/serializers/patient_consultation.py [hunk 1]
```diff
@@ -393,6 +393,9 @@ class PatientConsultationDischargeSerializer(serializers.ModelSerializer):
     discharge_reason = serializers.ChoiceField(
         choices=DISCHARGE_REASON_CHOICES, required=True
     )
+    referred_to_external = serializers.CharField(
+        required=False, allow_null=True, allow_blank=True
+    )
```
Comment (sainak) [inline care/facility/api/serializers/patient_consultation.py:398]: Explicitly specifying this is not required

**Comment by sainak** [comment]
@vigneshhari this is a good time to migrate and remove `referred_to` foreign key
https://github.com/coronasafe/care/blob/b0eefd152511a390f759b41dac0cb7e27c9ee117/care/facility/models/patient_consultation.py#L92-L98

File: care/facility/api/serializers/patient_consultation.py (hunk 1 above)
Comment (sachdevavaibhav) [reply care/facility/api/serializers/patient_consultation.py]: @sainak I have pushed the suggested changes. Kindly review it.

**Comment by sainak** [comment]
@sachdevavaibhav can you add some tests to validate whether this field is accepting correct inputs

**Comment by sachdevavaibhav** [comment]
> @sachdevavaibhav can you add some tests to validate whether this field is accepting correct inputs

ya sure!

**Comment by vigneshhari** [comment]
> @vigneshhari this is a good time to migrate and remove `referred_to` foreign key
> 
> https://github.com/coronasafe/care/blob/b0eefd152511a390f759b41dac0cb7e27c9ee117/care/facility/models/patient_consultation.py#L92-L98

This is more of an ops question @gigincg

**Comment by gigincg** [comment]
@sainak @vigneshhari And why is reffered_to deprecated?

**Review by sainak — APPROVED** [review]

**Comment by sainak** [comment]
> @sainak I have a couple of questions:
> - referred_to_external should accept only strings/null values right? And not integer etc.
Yes
> - Should I create separate branch for tests or push them here only?
You can include them in this pr

**Comment by sachdevavaibhav** [comment]
> > @sainak I have a couple of questions:
> > 
> > * referred_to_external should accept only strings/null values right? And not integer etc.
> 
> Yes
> 
> > * Should I create separate branch for tests or push them here only?
> 
> You can include them in this pr

@sainak I have written a few test and found that its also accepting integers (may be its converting it to string values). Kindly review the test and should I also add some validation to the input?

**Comment by sachdevavaibhav** [comment]
> > > @sainak I have a couple of questions:
> > > 
> > > * referred_to_external should accept only strings/null values right? And not integer etc.
> > 
> > 
> > Yes
> > > * Should I create separate branch for tests or push them here only?
> > 
> > 
> > You can include them in this pr
> 
> @sainak I have written a few test and found that its also accepting integers (may be its converting it to string values). Kindly review the test and should I also add some validation to the input?

@sainak

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**